fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java

```diff
@@ -975,8 +976,21 @@ public static Expression processInPredicate(InPredicate inPredicate) {
             }
             return inPredicate;
         }
+        // process string literal with numeric
+        AtomicReference<InPredicate> atomicInPredicate = new AtomicReference<>(inPredicate);
+        if (!(inPredicate.getCompareExpr().getDataType().isStringLikeType())) {
+            List<Expression> newOptions = inPredicate.getOptions().stream().map(option -> {
```

Comment:
do not use stream api, stream api hurt performance

Reply:
test case added